src/VOTables.jl

```diff
@@ -221,14 +221,16 @@ function _filltable!(cols, tblx, ::Val{:BINARY})
                 let
                     lenbytes = @view dataraw[i:i+4-1]
                     i += 4
-                    _parse_binary(Int32, lenbytes)
+                    _parse_binary(Int32, lenbytes) * TYPE_VO_TO_NBYTES[colspec[:datatype]]
                 end
             )
             curdata = @view dataraw[i:i+len-1]
             i += len
             value = _parse_binary(vo2jltype(colspec), curdata)
             if haskey(colspec, :nullvalue) && colspec[:nullvalue] == value
                 push!(col, missing)
+            elseif value isa AbstractArray && all(ismissing, value)
```

Comment:
At the moment this only happens for `_parse_binary(::Type{Bool},data)`, which can map some byte patterns to `missing`. `datatype='boolean'` is the only data type for which there exist standard fixed-length byte patterns (0x00, 0x20 or 0x3f) corresponding to a scalar value that can't be represented by the corresponding Julia data type (`Bool`) without use of `Missing`.

Arguably, array element values for other types corresponding to the `:nullvalue`, where present, should also be represented by `missing` elements, but I haven't implemented that.  The VOTable standard is a bit hazy on correct behaviour here, and parsers in some other languages tend not to do that because they don't have an efficient way to represent e.g. integers and missing values in the same array.